Submitty/Submitty
# [Feature:Submission] Expand/Collapse All Test Cases

### What is the current behavior?

- Feature #7866
  - Currently, there is no way to expand all gradeable test cases other than manually selecting each test case to open them.

### What is the new behavior?
- If a gradeable has at least one test case that can display additional details, then an "Expand All Test Cases" toggle and a "Collapse All Test Cases" toggle will be visible above the test cases.
- When selected, the "Expand All Test Cases" toggle will open all test cases.
- When selected, the "Collapse All Test Cases" toggle will close all test cases.
- If all test cases are opened (either through the toggle or manually), the "Expand All Test Cases" toggle's text will be greyed out until one test case is closed.
- If all test cases are closed (either through the toggle or manually), the "Collapse All Test Cases" toggle's text will be greyed out until one test case is opened.

### Other information?
**How did you test**
In a gradeable that has a couple of test cases with additional details (i.e. `sample > Leaderboard` or `sample > Autograder Hidden and Extra Credit (C++ Hidden Tests)` ) : 
 - Select the "Expand All Test Cases" toggle and check that all test cases are successfully opened.
 - Select the "Collapse All test Cases" toggle and check that all test cases successfully close.
 - Select the "Expand All Test Cases" toggle, and as the test cases are loading, select "Collapse All test Cases".  Check that all test cases were aborted.
 - Manually open each test case in the gradeable and check that the "Expand All Test Cases" is greyed out and the "Collapse All Test Cases" is interactable.
 - Manually close each test case in the gradeable and check that the "Collapse All Test Cases" is greyed out and the "Expand All Test Cases" is interactable.
 - Manually open each test case in the gradeable then as they are loading, cancel each of the test case from loading.  Check that the "Collapse All Test Cases" is greyed out and the "Expand All Test Cases" is interactable.  This was tested on `sample > Leaderboard` as there are only two test cases in this gradeable.

In a gradeable that does not have a test case, check that the "Expand/Collapse All Test Cases" toggles are not displayed.


**Screenshots**
![image](https://user-images.githubusercontent.com/20442222/183094630-56c84007-92c8-4b40-85d7-3591ab143e3e.png)

![image](https://user-images.githubusercontent.com/20442222/183094465-28049ec9-0414-49b5-a4cd-d472df67bc5e.png)

![image](https://user-images.githubusercontent.com/20442222/183100899-3f45f09b-960d-466f-8d5c-731ecececb20.png)
